MdeModulePkg/Universal/Console/TerminalDxe/TerminalConIn.c

```diff
@@ -95,6 +95,10 @@ TerminalConInReset (
       );
   }
 
+  if (!EFI_ERROR (Status)) {
+    Status = TerminalDevice->SerialIo->SetControl (TerminalDevice->SerialIo, EFI_SERIAL_DATA_TERMINAL_READY|EFI_SERIAL_REQUEST_TO_SEND);

```

Comment:
SerialIo->Reset() is called above.  Should that API reset this state so it is common to all consumers of the Serial I/O Protocol?

Reply:
I do see one Serial I/O Protocol implementation preserves the control settings across Reset():

https://github.com/tianocore/edk2/blob/b158dad150bf02879668f72ce306445250838201/MdeModulePkg/Bus/Pci/PciSioSerialDxe/SerialIo.c#L643

The question is if this is the correct behavior for Serial I/O Protocol.  UEFI Spec description of Serial I/O Reset() does not cover this topic.

Reply:
Hi Mike,

The Serial I/O just says that it resets the hardware, without details of what that means.

In this instance, in the Terminal Console In Devices, when serial I/O is reset, it is with the intention of being able to consume data from the device as a Simple Text In protocol. 

Most serial devices, on reset, will have the DTR and RTS bits default enabled, which signals an attached device that data can now be sent. But the Uefi code does not make it explicit that DTR and RTS should be set.

This PR is handing the case that if a serial device is expected to be used as a ConIn device, that the device is configured to start receiving data.

Reply:
What about the console output side of TerminalDxe.  Would that have the same issue?

Reply:
In the hardware we found this on, con out was working no matter the state of dtr/rts.  It's why this is being scoped to only the con in side.